### PR TITLE
Marco/courses route

### LINKED ISSUE #17: As an Admin, I would like to C.R.U.D Courses so that I am able to keep the courses in the database up to date.
As an Admin, I would like to C.R.U.D Courses so that I am able to keep the courses in the database up to date.

### **Acceptance criteria**:
- [x] Create course route
- [x] Read course route
- [x] Update course route
- [x] Delete course route

### PR DESCRIPTION
**Pull request by marcoarceo**
Made updates to the courses routes so that they can be up to date

This PR closes #17

### CODE REVIEW

**Review by rebeccarothschild — APPROVED** [review]
Good job

File: server/routes/course.js [hunk 1]
```diff
@@ -10,114 +10,113 @@ const User = require('./../schemas/user');
...
+      !req.body.hasOwnProperty("courseCurrentlyEnrolled") || 
+      !req.body.hasOwnProperty("courseNotes")) {
+    //Body does not contain correct properties
+    return res.status(400).send("POST request on /courses formulated incorrectly." +
+      "Body must contain all 10 required fields: courseInstructorFirstName, courseInstructorLastName, courseInstructorID, courseName, courseNumber, courseYear, courseSemester, courseEnrollmentLimit, courseCurrentlyEnrolled, courseNotes");
+  }
+  try {
+    let status = await User.updateOne(
+    {id: req.params.userId},
+    {$push: {courses: req.body}});
```
Comment (rebeccarothschild) [inline server/routes/course.js:35]: Actually @marcoarceo as I was working on students CRUD, I realized that we need to add a courseId to the body before we push it. Could you add code for this? courseId could be a combo of course number, year and semester.

Comment (marcoarceo) [reply server/routes/course.js:35]: Yes, I can add the courseID by tomorrow morning. We would have to set the courseID after the user adds the information required to create a course. Before calling the route to POST, we would assign courseID to a combination of courseName + courseNumber + courseSemester + courseYear.

Comment (rebeccarothschild) [reply server/routes/course.js:35]: Sounds good! No rush, I can manually add it in the meantime

Comment (marcoarceo) [reply server/routes/course.js:35]: Added the courseID to the body, please review the code and let me know if that was what you asked for.

**Review by rebeccarothschild — APPROVED** [review]

**Review by HatchedCode — APPROVED** [review]
Look go to go!

**Pull request merged**